s3fs/core.py

```diff
@@ -97,6 +98,8 @@ class S3FileSystem(object):
     use_ssl : bool (True)
         Whether to use SSL in connections to S3; may be faster without, but
         insecure
+    writer_kwargs : dict of parameters that are used when writing data to s3.
+        Typically used for things like "ServerSideEncryption".
```

Comment:
So this should be "file_kwargs" and I would put here that they can be over-ridden by kwargs to any of the methods which accept them.